robot/basestation/static/js/stream.js

```diff
@@ -0,0 +1,33 @@
+$(document).ready(function () {
+    $('#arm-stream-btn').on('click', function (event) {
+        // click makes it checked during this time, so trying to enable
+        if ($('#arm-stream-btn').is(':checked')) {
+            $('img#arm-camera-feed')[0].src =
+            'http://localhost:8080/stream?topic=/ArmCamera/image_raw'
+        } else {
+            $('img#arm-camera-feed')[0].src =
+                '../static/images/stream-offline.jpg'
+        }
+    })
+    $('#front-stream-btn').on('click', function (event) {
+        // click makes it checked during this time, so trying to enable
+        if ($('#front-stream-btn').is(':checked')) {
+            $('img#front-camera-feed')[0].src =
+            'http://localhost:8080/stream?topic=/FrontCamera/image_raw'
+        } else {
+            $('img#front-camera-feed')[0].src =
+                '../static/images/stream-offline.jpg'
+        }
+    })
+
+    $('#rear-stream-btn').on('click', function (event) {
+        // click makes it checked during this time, so trying to enable
+        if ($('#rear-stream-btn').is(':checked')) {
+            $('img#rear-camera-feed')[0].src =
+            'http://localhost:8080/stream?topic=/RearCamera/image_raw'
+        } else {
+            $('img#rear-camera-feed')[0].src =
+                '../static/images/stream-offline.jpg'
+        }
+    })
```

Comment:
This is code duplication. I suggest creating a function of this signature

`string getStreamURL(streamName)`
That returns 'http://localhost:8080/stream?topic=/' +streamName +' /image_raw'

Also put the '../static/images/stream-offiline.jpg' in a constant.

then you have at maximum
```
$('#rear-stream-btn).click(function(){
 if ($('#rear-stream-btn').is(':checked')) {
            $('img#rear-camera-feed')[0].src = getStreamURL('RearCamera')
        } else {
            $('img#rear-camera-feed')[0].src = CONSTANT
        }
})
```

I would also remove the comment as it's a bit confusing, reading the code is much simpler. What does it try to say by click makes it checked _during this time_ (What time?), so trying to enable??

Reply:
Agreed, fixed